sections/dashboard/Stake/InputCards/StakeInputCard.tsx

```diff
@@ -3,99 +3,103 @@ import _ from 'lodash';
 import { FC, useCallback, useMemo, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import styled from 'styled-components';
-import { useContractWrite, usePrepareContractWrite } from 'wagmi';
 
 import Button from 'components/Button';
 import NumericInput from 'components/Input/NumericInput';
 import SegmentedControl from 'components/SegmentedControl';
 import { DEFAULT_CRYPTO_DECIMALS, DEFAULT_TOKEN_DECIMALS } from 'constants/defaults';
-import { monitorTransaction } from 'contexts/RelayerContext';
-import { useStakingContext } from 'contexts/StakingContext';
-import { STAKING_LOW_GAS_LIMIT } from 'queries/staking/utils';
-import { truncateNumbers, zeroBN } from 'utils/formatters/number';
+import { useAppDispatch, useAppSelector } from 'state/hooks';
+import { approveKwentaToken } from 'state/staking/actions';
+import { stakeKwenta, unstakeKwenta } from 'state/staking/actions';
+import {
+	selectIsKwentaTokenApproved,
+	selectKwentaBalance,
+	selectStakedKwentaBalance,
+} from 'state/staking/selectors';
+import { toWei, truncateNumbers } from 'utils/formatters/number';
 
 import { StakingCard } from '../common';
 
 const StakeInputCard: FC = () => {
 	const { t } = useTranslation();
-	const {
-		kwentaBalance,
-		stakedNonEscrowedBalance,
-		kwentaApproveConfig,
-		kwentaTokenApproval,
-		stakingRewardsContract,
-	} = useStakingContext();
+	const dispatch = useAppDispatch();
+
+	const kwentaBalance = useAppSelector(selectKwentaBalance);
+	const stakedKwentaBalance = useAppSelector(selectStakedKwentaBalance);
 
 	const [amount, setAmount] = useState('');
 	const [activeTab, setActiveTab] = useState(0);
-	const amountBN = useMemo(
-		() => (amount === '' ? zeroBN.toString(0, true) : wei(amount).toString(0, true)),
-		[amount]
-	);
+	const amountBN = useMemo(() => toWei(amount).toString(0, true), [amount]);
```

Comment:
Again here maybe we want `wei(amount).toBN()`

Reply:
Done.